### PR TITLE
Improve telemetry

### PR DESCRIPTION
**Pull request by michaljurecko**
Slack: https://keboola.slack.com/archives/CL11GKH6H/p1675140020514929?thread_ts=1675093006.557839&cid=CL11GKH6H

### CODE REVIEW

File: internal/pkg/service/buffer/worker/task/node.go [hunk 1]
```diff
@@ -134,9 +140,22 @@ func (n *Node) StartTask(ctx context.Context, exportKey key.ExportKey, typ, lock
...
+	span.SetAttributes(
+		telemetry.KeepSpan(),
+		attribute.String("projectId", task.ProjectID.String()),
+		attribute.String("receiverId", task.ReceiverID.String()),
+		attribute.String("exportId", task.ExportID.String()),
+		attribute.String("lock", task.Lock),
+		attribute.String("worker", task.WorkerNode),
+		attribute.String("createdAt", task.CreatedAt.String()),
+	)
+
```
Comment (michaljurecko) [inline internal/pkg/service/buffer/worker/task/node.go:158]: Pridal som rovno telemetriu pre buffer tasky.

File: internal/pkg/service/templates/api/http/middleware.go [hunk 2]
```diff
@@ -31,21 +31,21 @@ func TraceEndpointsMiddleware(serverDeps dependencies.ForServer) func(endpoint g
...
+				tracer.ResourceName(resourceName),
+			}
+
 			// Trace all endpoints except health check
-			analyticRate := 1.0
 			if strings.Contains(resourceName, "HealthCheck") {
-				analyticRate = 0.0
+				opts = append(opts, tracer.AnalyticsRate(0.0), tracer.Tag(ext.ManualDrop, true))
+			} else {
+				opts = append(opts, tracer.AnalyticsRate(1.0), tracer.Tag(ext.ManualKeep, true))
```
Comment (michaljurecko) [inline internal/pkg/service/templates/api/http/middleware.go:43]: A tu je hlavna zmena, `endpoint` spans su explicine nastavene na drop pre health check a keep pre ostatne.

File: internal/pkg/service/buffer/worker/service/slice_retry_test.go [hunk 3]
```diff
@@ -108,8 +109,12 @@ func TestRetryFailedUploadsTask(t *testing.T) {
 	workerDeps.DebugLogger().Truncate()
 	clk.Add(3 * time.Minute)
 
+	// Wait for failed upload and the retry
+	assert.Eventually(t, func() bool {
+		return strings.Count(workerDeps.DebugLogger().WarnMessages(), "WARN  task failed") == 1
+	}, 10*time.Second, 100*time.Millisecond)
+
```
Comment (michaljurecko) [inline internal/pkg/service/buffer/worker/service/slice_retry_test.go:116]: Vylepseny test.

File: internal/pkg/service/buffer/worker/distribution/nodeconfig.go [hunk 4]
```diff
@@ -7,7 +7,7 @@ import (
 const (
 	DefaultSessionTTL          = 15               // seconds, see WithTTL
 	DefaultStartupTimeout      = 60 * time.Second // timeout for registration, PUT operation
-	DefaultShutdownTimeout     = 5 * time.Second  // timeout for un-registration, DELETE operation
+	DefaultShutdownTimeout     = 10 * time.Second // timeout for un-registration, DELETE operation
```
Comment (michaljurecko) [inline internal/pkg/service/buffer/worker/distribution/nodeconfig.go:10]: V CI to nestacilo.

**Review by JakubMatejka — APPROVED** [review]

**Pull request merged**